agent/engine/docker_image_manager.go

```diff
@@ -277,6 +275,8 @@ func (imageManager *dockerImageManager) performPeriodicImageCleanup(ctx context.
 }
 
 func (imageManager *dockerImageManager) removeUnusedImages() {
+	imageManager.updateLock.Lock()
```

Comment:
Github does not let me add comments on code that needs to modified outside this PR, so commenting here instead. It'd be a lot better if we could add some logging here because this method acquires at least 3 locks during its lifetime. Specifically, a log line to indicate that we are cleaning all tracking information for the image name because it has 0 references to it at https://github.com/aws/amazon-ecs-agent/pull/743/files#diff-beabb53f72ea56fdfd51386c12eaf9bfR351
2.